t-rex-core/src/cache/s3cache_test.rs

```diff
@@ -8,9 +8,9 @@ use curl::easy::Easy;
 use std::env;
 use std::str;
 
-#[test]
+#[tokio::test]
 #[ignore]
-fn test_s3cache() {
+async fn test_s3cache() {
     if env::var("S3TEST").is_err() {
```

Comment:
Does this test work for you?

Gitlab CI seems to hang here:
```
2021-03-30T14:33:39.1717639Z test cache::s3cache_test::test_s3cache ... test cache::s3cache_test::test_s3cache has been running for over 60 seconds
2021-03-30T20:25:50.2145087Z ##[error]The operation was canceled.
```

Reply:
Yeah, it hangs on CI. I didn't try it locally yet, I'll have to look into setting up minio. On the other hand, it might make sense to make the functions return a future, which could solve this.